get inbox messages from Accengage with pagination

edaluz opened the pull request.

Commit:
Merge branch 'release/1.1.4'

emilianoMonk2017 committed:
ENUMs added to RNAAccengageModule header for error cases

emilianoMonk2017 committed:
- getAccengageInbox added to RNAAccengageModule
- getAccengageMessages added to RNAAccengageModule

emilianoMonk2017 committed:
- getInboxMessages added to index.js
- getMessage added to index.js

emilianoMonk2017 committed:
paginator added to RNAccengageModule for request for  Accengage Messages

emilianoMonk2017 committed:
Merge branch 'master' of https://github.com/edaluz/react-native-accengage

# Conflicts:
#	ios/RNAccengageModule/RNAccengageModule.m

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -8,11 +8,16 @@
 static NSString *const kPushRequested = @"pushRequested";
 
 @implementation RNAccengageModule
+    BMA4SInBox      *_inboxMessageList;
```
I'd call this _inbox, as it is not a list, it has more functionality.

edaluz replied:
It was renamed to "inbox"

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+{
+    _messageList = [NSMutableArray new];
+    
+    if(_numLoadedMessages < _inboxMessageList.size)
```
There should be validation to check in _inbox is existing.

edaluz replied:
This validations is inside the method get Accengage inbox. After the inbox is returned with the success callback, it calls to the get messages method.

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+    
+    if(_numLoadedMessages < _inboxMessageList.size)
+    {
+        int unreadMessagesCount = (int)(_inboxMessageList.size - _numLoadedMessages);
```
This has nothing to do with messages being read or unread right?

erikpoort replied:
Why is _numLoadedMessages subtracted here?

edaluz replied:
I Updated for this:
     int unshownMessages = (int)(_inbox.size - _numLoadedMessages);
     int batchSize = _numLoadedMessages + MIN(unshownMessages, 10);

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+        for (int i = 0; i < _numLoadedMessages; i++)
+        {
+            [_messageList addObject:[NSNull null]];
+            [self getInboxMessageAtIndex:i+1 messageCallback:^(NSArray *response) {
```
Why i+1?

edaluz replied:
It was a test because and I forgot to remove it

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+                [_messageList setObject:response.firstObject atIndexedSubscript:i];
+                
+                //if the message is the last of the array
+                if(_messageList.count == i)
```
This will always trigger when the first message is loaded, i will be the _numLoadedMessages at that point. And _messageList.count will always be that amount as well.

edaluz replied:
I changed this for this:
              if(batchSize - 1 == i)

and the batchSize is:
        int unshownMessages = (int)(_inbox.size - _numLoadedMessages);
        int batchSize = _numLoadedMessages + MIN(unshownMessages, 10);

erikpoort commented on ios/RNAccengageModule/RNAccengageModule.m:
```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+                                          @"sender"      : message.from,
+                                          @"read"        : [NSNumber numberWithBool:message.isRead],
+                                          @"archived"    : [NSNumber numberWithBool:message.isArchived]
+                                          };
```
customParameters are missing here

edaluz replied:
I added to the list:
//Create Message Dictionary
            NSDictionary *messageData = @{@"title"       : message.title,
                                          @"body"        : message.text,
                                          @"timestamp"   : message.date,
                                          @"category"    : message.category,
                                          @"sender"      : message.from,
                                          @"read"        : [NSNumber numberWithBool:message.isRead],
                                          @"archived"    : [NSNumber numberWithBool:message.isArchived],
                                          @"customParameters" : message.customParams
                                          };

erikpoort commented:
I have added some comments to your code!
Also, there's no pagination in here at all.
Also, can you base the PR on develop?

emilianoMonk2017 committed:
- inboxMessageList renamed to inbox
- get AccengageMessages between  numLoadedMessages and  batchSize if the  number of loaded messages is smaller than inbox's size
- customParameters added to message dictionary

emilianoMonk2017 committed:
Merge branch 'master' of https://github.com/edaluz/react-native-accengage

# Conflicts:
#	ios/RNAccengageModule/RNAccengageModule.m

emilianoMonk2017 committed:
adaption of the IOS project to the Android version

emilianoMonk2017 committed:
Merge branch 'master' of https://github.com/edaluz/react-native-accengage

# Conflicts:
#	ios/RNAccengageModule/RNAccengageModule.h
#	ios/RNAccengageModule/RNAccengageModule.m
#	src/index.js

emilianoMonk2017 committed:
- get dummy with messages added to RNAccengageModule
- RCTResponseSenderBlock updated to RCTPromiseResolveBlock in all inbox methods
- error message dictionary added to the message list when the get message call had failed
